client/task-list/tasks/payments/index.js

```diff
@@ -37,15 +37,24 @@ class Payments extends Component {
 		const { methods } = props;
 
 		const enabledMethods = {};
-		methods.forEach(
-			( method ) => ( enabledMethods[ method.key ] = method.isEnabled )
-		);
+		let recommendedMethod = 'stripe';
```

Comment:
Was there a reason for moving this back to the constructor?  I hit the bug this time where WC Pay was not in the list of payment methods.

I think the method to check for visible/enabled and updating recommended should be moved to its own function and called in the constructor and in `componentDidUpdate()`.

Reply:
> Was there a reason for moving this back to the constructor?

I had only re-calculated `recommendedMethod` on `componentDidUpdate()`, that was a mistake. WCPay doesn't require Jetpack anymore, so recalculating `recommendedMethod` isn't needed at all...

> I hit the bug this time where WC Pay was not in the list of payment methods.

... or so I thought. Anyway, your solution is better :)